sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs

```diff
@@ -242,6 +242,21 @@ protected PartitionReceiver()
             OwnsConnection = false;
         }
 
+        /// <summary>
+        ///   Retrieves information about the partition this client is associated to, including elements that describe the
+        ///   available events in the partition event stream.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information for the associated partition under the Event Hub this client is associated with.</returns>
+        ///
+        public virtual Task<PartitionProperties> GetPartitionPropertiesAsync(CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(PartitionReceiver));
+            return Connection.GetPartitionPropertiesAsync(PartitionId, RetryPolicy, cancellationToken);
```

Comment:
It has been recommended that we `await` these rather than returning the raw task.   _(see: [David Fowler's guidance](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task))_.

Reply:
Actually that's what I was asking about. I'll update other clients as well.